packages/ember-routing-htmlbars/lib/keywords/closure-action.js

```diff
@@ -45,6 +45,12 @@ export default function closureAction(morph, env, scope, params, hash, template,
         if (!action) {
           throw new EmberError(`An action named '${actionName}' was not found in ${target}.`);
         }
+      } else if (actionType === 'function') {
+        // on-change={{action setName}}
```

Comment:
This isn't the correct code-path. Note that the `actionType` here is `'function'`. The description in your PR is that you want `(action actionName target=otherObject)` to work, which it does:

* http://emberjs.jsbin.com/kuxuhukuyu/1/edit?html,js,output

This code and the test are demonstrating a different case: `(action someFunc target=newThisBinding)`. Of course if `someFunc` was already bound (as it is during normal use via actions) then specifying a target would do nothing.